### PR TITLE
Fix leaking goroutines in multiple integration tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

Note to reviewer: I recommend reviewing commits one by one - they are all independent.

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig testing

### CODE REVIEW

**Comment by aojea** [comment]
job failures seems related

File: test/integration/quota/quota_test.go [hunk 1]
```diff
@@ -64,41 +57,20 @@ const (
...
-	qca := quotainstall.NewQuotaConfigurationForAdmission()
-
-	initializers := admission.PluginInitializers{
-		genericadmissioninitializer.New(clientset, internalInformers, nil, nil, admissionCh),
-		kubeapiserveradmission.NewPluginInitializer(nil, nil, qca),
-	}
-	initializers.Initialize(admissionControl)
-	if err := admission.ValidateInitialization(admissionControl); err != nil {
-		t.Fatalf("couldn't initialize resource quota: %v", err)
-	}
```
Comment (aojea) [inline test/integration/quota/quota_test.go:91]: no longer needed?

File: test/integration/apiserver/apiserver_test.go [hunk 2]
```diff
@@ -64,52 +64,38 @@ import (
...
 	kubeapiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
-
 	"k8s.io/kubernetes/pkg/controlplane"
 	"k8s.io/kubernetes/test/integration"
 	"k8s.io/kubernetes/test/integration/etcd"
 	"k8s.io/kubernetes/test/integration/framework"
 )
 
-func setup(t testing.TB, groupVersions ...schema.GroupVersion) (*httptest.Server, clientset.Interface, framework.CloseFunc) {
+func setup(t *testing.T, groupVersions ...schema.GroupVersion) (clientset.Interface, *restclient.Config, framework.TearDownFunc) {
```
Comment (aojea) [inline test/integration/apiserver/apiserver_test.go:75]: finally :tada:

Comment (aojea) [inline test/integration/apiserver/apiserver_test.go:75]: finally :tada:

File: test/integration/apimachinery/watch_restart_test.go [hunk 3]
```diff
@@ -66,19 +66,16 @@ func TestWatchRestartsIfTimeoutNotReached(t *testing.T) {
 	// Has to be longer than 5 seconds
 	timeout := 30 * time.Second
 
-	// Set up an API server
-	controlPlaneConfig := framework.NewIntegrationTestControlPlaneConfig()
-	// Timeout is set random between MinRequestTimeout and 2x
-	controlPlaneConfig.GenericConfig.MinRequestTimeout = int(timeout.Seconds()) / 4
-	_, s, closeFn := framework.RunAnAPIServer(controlPlaneConfig)
-	defer closeFn()
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--min-request-timeout=1"}, framework.SharedEtcd())
```
[the remark this answers is not in the document]
Comment (wojtek-t) [reply test/integration/apimachinery/watch_restart_test.go]: good catch. Switched to 7 (as the flag accepts only integer seconds).

File: test/integration/quota/quota_test.go (hunk 1 above)
Comment (wojtek-t) [reply test/integration/quota/quota_test.go:91]: Yes - when using the unified way of setting the apiserver, resource quota admission is just enabled there - so we only need to configure it correctly (and this particular test is using default configuration, the other two below are using  the custom one so we need to create that config file).

File: test/integration/quota/quota_test.go [hunk 4]
```diff
@@ -292,49 +263,43 @@ func scale(t *testing.T, namespace string, clientset *clientset.Clientset) {
...
+plugins:
+- name: ResourceQuota
+  configuration:
+    apiVersion: apiserver.config.k8s.io/v1
+    kind: ResourceQuotaConfiguration
+    limitedResources:
+    - resource: pods
+      matchContains:
+      - pods
+`), os.FileMode(0755)); err != nil {
```
[the remark this answers is not in the document]
Comment (wojtek-t) [reply test/integration/quota/quota_test.go]: Correct - copy paster error. Fixed.

File: test/integration/quota/quota_test.go [hunk 5]
```diff
@@ -425,50 +389,43 @@ func TestQuotaLimitedResourceDenial(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (wojtek-t) [reply test/integration/quota/quota_test.go]: Correct - copy paster error. Fixed.

**Review by wojtek-t — COMMENTED** [review]
@aojea - thanks for the review

PTAL

**Comment by fedebongio** [comment]
Thank you @wojtek-t !
/triage accepted

**Pull request merged**